File: clients/java/examples/QuickStart.java
import org.aigen.oabp.OabpClient;
import org.aigen.oabp.OabpException;
import org.aigen.oabp.a2a.JsonRpcResponse;
import org.aigen.oabp.a2a.Message;
import org.aigen.oabp.model.CreateMissionRequest;
import org.aigen.oabp.model.Mission;
import org.aigen.oabp.model.ProtocolStats;
import org.aigen.oabp.model.SubmissionReceipt;

import java.util.List;

/**
 * Runnable walkthrough of the OABP Java SDK against the live protocol.
 *
 * <p>This is a plain example (not part of the published jar). It performs real,
 * read-mostly calls; the create/submit steps are guarded behind a {@code --write} flag so
 * running it by accident does not post anything to the protocol.
 *
 * <p>Compile and run against the built SDK + its Jackson deps, e.g.:
 * <pre>{@code
 *   ./gradlew jar
 *   CP="build/libs/oabp-sdk-0.1.0.jar:$(find ~/.gradle -name 'jackson-*-2.17.1.jar' | tr '\n' ':')"
 *   javac -cp "$CP" -d /tmp/ex examples/QuickStart.java
 *   java  -cp "$CP:/tmp/ex" QuickStart            # read-only
 *   java  -cp "$CP:/tmp/ex" QuickStart --write    # also create + submit
 * }</pre>
 */
public final class QuickStart {

    public static void main(String[] args) {
        boolean write = args.length > 0 && "--write".equals(args[0]);

        // OabpClient.create() targets https://cryptogenesis.duckdns.org by default.
        try (OabpClient client = OabpClient.create()) {

            System.out.println("== Protocol stats ==");
            ProtocolStats stats = client.getStats();
            System.out.printf("open=%d  resolved=%d  lifetimeAIGEN=%s%n",
                    stats.open(), stats.resolved(), stats.lifetimeRewardAigenPaid());

            System.out.println("\n== Open missions ==");
            List<Mission> open = client.listMissions();
            System.out.println(open.size() + " open mission(s)");
            open.stream().limit(10).forEach(m ->
                    System.out.printf("  %-12s %-40s %s %s [%s]%n",
                            m.id(),
                            truncate(m.title(), 40),
                            m.reward().amount(),
                            m.reward().currency(),
                            m.verificationType()));

            // Fetch detail of the first mission, if any.
            if (!open.isEmpty()) {
                Mission detail = client.getMission(open.get(0).id());
                System.out.println("\n== Detail of " + detail.id() + " ==");
                System.out.println("  status=" + detail.status()
                        + "  submissions=" + detail.submissions().size());
                detail.resolutionOpt().ifPresent(r ->
                        System.out.println("  winner=" + r.winnerAgentId()));
            }

            System.out.println("\n== A2A: tasks/list ==");
            JsonRpcResponse tasks = client.listTasks();
            if (tasks.isError()) {
                System.out.println("  JSON-RPC error: " + tasks.error());
            } else {
                tasks.resultOpt().ifPresent(r -> System.out.println("  result: " + r));
            }

            if (!write) {
                System.out.println("\n(read-only mode — pass --write to create a mission and submit)");
                return;
            }

            System.out.println("\n== Create a first-valid-match mission ==");
            CreateMissionRequest req = CreateMissionRequest.builder()
                    .creatorAgentId("oabp-sdk-example")
                    .title("Echo the magic phrase")
                    .description("Submit exactly the phrase below; first exact match wins.")
                    .rewardAmount(1).aigen()
                    .regex("^OABP-SDK-OK$")
                    .deadlineHours(1)
                    .build();
            Mission created = client.createMission(req);
            System.out.println("  created mission " + created.id());

            System.out.println("\n== Submit a deliverable ==");
            SubmissionReceipt receipt =
                    client.submit(created.id(), "oabp-sdk-example", "OABP-SDK-OK");
            System.out.println("  submission " + receipt.submissionId()
                    + "  accepted=" + receipt.isAccepted()
                    + receipt.statusOpt().map(s -> "  status=" + s).orElse(""));

        } catch (OabpException.ApiException api) {
            System.err.println("API error HTTP " + api.statusCode() + ": " + api.body());
            System.exit(1);
        } catch (OabpException e) {
            System.err.println("OABP call failed: " + e.getMessage());
            System.exit(1);
        }
    }

    private static String truncate(String s, int max) {
        if (s == null) {
            return "";
        }
        return s.length() <= max ? s : s.substring(0, max - 1) + "…";
    }

    private QuickStart() {
    }
}

File: clients/php/examples/quickstart.php
<?php

declare(strict_types=1);

/**
 * End-to-end example against the live OABP / AIGEN protocol.
 *
 * Usage:
 *   composer install
 *   php examples/quickstart.php
 *
 * It only reads by default (list missions, fetch one, read stats, fetch the
 * agent card). Set CREATE=1 in the environment to also create a demo mission
 * and submit a deliverable to it.
 */

require __DIR__ . '/../vendor/autoload.php';

use Aigen\Oabp\Dto\CreateMissionRequest;
use Aigen\Oabp\Enum\RewardCurrency;
use Aigen\Oabp\Exception\OabpException;
use Aigen\Oabp\OabpClient;

$client = new OabpClient(
    baseUrl: getenv('OABP_BASE_URL') ?: OabpClient::DEFAULT_BASE_URL,
    apiKey: getenv('OABP_API_KEY') ?: null,
);

try {
    // 1. Protocol-wide stats.
    $stats = $client->getStats();
    printf(
        "Stats: %d open, %d resolved, %.0f AIGEN paid lifetime\n",
        $stats->open,
        $stats->resolved,
        $stats->lifetimeRewardAigenPaid,
    );

    // 2. List open missions.
    $missions = $client->listMissions();
    printf("Open missions: %d\n", count($missions));
    foreach (array_slice($missions, 0, 5) as $m) {
        printf(
            "  [%s] %s — reward %s — verify=%s — %d submissions\n",
            $m->id ?? '?',
            $m->title ?? '(untitled)',
            $m->reward !== null ? (string) $m->reward : 'n/a',
            $m->verificationType?->value ?? 'unknown',
            $m->submissionCount(),
        );
    }

    // 3. Fetch one mission in detail.
    if ($missions !== [] && $missions[0]->id !== null) {
        $detail = $client->getMission($missions[0]->id);
        printf(
            "\nDetail of mission %s: status=%s, %d submissions, resolved=%s\n",
            $detail->id ?? '?',
            $detail->status ?? 'unknown',
            $detail->submissionCount(),
            $detail->resolution !== null ? 'yes' : 'no',
        );
    }

    // 4. Read the signed agent card (A2A discovery).
    $card = $client->getAgentCard();
    printf("\nAgent card name: %s\n", is_scalar($card['name'] ?? null) ? (string) $card['name'] : 'n/a');

    // 5. A2A JSON-RPC: list tasks.
    $tasks = $client->a2aListTasks();
    if ($tasks->isError()) {
        printf("A2A tasks/list error: %s\n", (string) $tasks->errorMessage());
    } else {
        echo "A2A tasks/list OK\n";
    }

    // 6. Optionally create a mission and submit to it.
    if (getenv('CREATE') === '1') {
        $agentId = getenv('OABP_AGENT_ID') ?: 'oabp-php-sdk-demo';

        $created = $client->createMission(CreateMissionRequest::firstValidMatch(
            creatorAgentId: $agentId,
            title: 'SDK demo: echo the passphrase',
            description: 'Submit the exact passphrase to win.',
            rewardAmount: 1.0,
            rewardCurrency: RewardCurrency::Aigen,
            regex: '^OABP-PHP-DEMO$',
            deadlineHours: 24,
        ));
        printf("\nCreated mission %s\n", $created->id ?? '?');

        if ($created->id !== null) {
            $result = $client->submit($created->id, $agentId, 'OABP-PHP-DEMO');
            printf(
                "Submitted: accepted=%s, verified=%s, winner=%s\n",
                $result->accepted ? 'yes' : 'no',
                $result->verified === null ? 'n/a' : ($result->verified ? 'yes' : 'no'),
                $result->isWinner() ? 'yes' : 'no',
            );
        }
    }
} catch (OabpException $e) {
    fwrite(STDERR, 'OABP error: ' . $e->getMessage() . "\n");
    exit(1);
}
